codec-http/src/main/java/io/netty/handler/codec/http/cookie/ClientCookieDecoder.java

```diff
@@ -241,13 +244,20 @@ private void parse7(int nameStart, int valueStart, int valueEnd) {
             }
         }
 
-        private void parse8(int nameStart) {
+        private void parse8(int nameStart, int valueStart, int valueEnd) {
             if (header.regionMatches(true, nameStart, CookieHeaderNames.HTTPONLY, 0, 8)) {
                 httpOnly = true;
+            } else if (header.regionMatches(true, nameStart, CookieHeaderNames.SAMESITE, 0, 8)) {
+                String sameSiteValue = computeValue(valueStart, valueEnd);
+                try {
+                    sameSite = SameSite.of(sameSiteValue);
+                } catch (IllegalArgumentException e) {
+                    logger.debug("Skipping SameSite attribute because value '{}' is invalid", sameSiteValue);
+                }
             }
         }
 
-        private static boolean isValueDefined(int valueStart, int valueEnd) {
+        private boolean isValueDefined(int valueStart, int valueEnd) {
```

Comment:
Please revert to static. That's a general practice in Netty for methods that don't depend on any instance state.

Reply:
@dvlato ^^